packages/connectors/connector-yunpian-sms/src/index.ts

```diff
@@ -0,0 +1,128 @@
+import { assert } from '@silverhand/essentials';
+import { got, RequestError } from 'got';
+
+import type {
+  GetConnectorConfig,
+  SendMessageFunction,
+  CreateConnector,
+  SmsConnector,
+} from '@logto/connector-kit';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  validateConfig,
+  ConnectorType,
+  replaceSendMessageHandlebars,
+} from '@logto/connector-kit';
+
+import { defaultMetadata, endpoint } from './constant.js';
+import {
+  yunpianSmsConfigGuard,
+  type YunpianSmsPayload,
+  yunpianErrorResponseGuard,
+} from './types.js';
+
+// 验证中国手机号格式
+const isChinaPhoneNumber = (phoneNumber: string) => {
+  // 去除所有空格
+  const phone = phoneNumber.replaceAll(/\s/g, '');
+
+  // 匹配格式:
+  // 1. +86 后跟11位数字，第一位必须是1
+  // 2. 86 后跟11位数字，第一位必须是1
+  const pattern = /^(\+?86)1[3-9]\d{9}$/;
+
+  return pattern.test(phone);
+};
+
+// 处理手机号格式
+const formatPhoneNumber = (phoneNumber: string) => {
+  const phone = phoneNumber.replaceAll(/\s/g, '');
+
+  if (!isChinaPhoneNumber(phone)) {
+    if (!phone.startsWith('+')) {
+      return `+${phone}`;
+    }
+    return phone;
+  }
+
+  // 如果以 +86 或 86 开头，截取后11位
+  if (phone.startsWith('+86')) {
+    return phone.slice(3);
+  }
+  if (phone.startsWith('86')) {
+    return phone.slice(2);
+  }
+
+  return phone;
+};
+
+const sendMessage =
+  (getConfig: GetConnectorConfig): SendMessageFunction =>
+  async (data, inputConfig) => {
+    const { to, type, payload } = data;
+    const config = inputConfig ?? (await getConfig(defaultMetadata.id));
+    validateConfig(config, yunpianSmsConfigGuard);
+    const { apikey, enableInternational, templates } = config;
+
+    const template = templates.find((template) => template.usageType === type);
+    assert(
+      template,
+      new ConnectorError(ConnectorErrorCodes.TemplateNotFound, `找不到类型为 ${type} 的短信模板`)
+    );
+
+    const messageContent = replaceSendMessageHandlebars(template.content, payload);
+
+    const formattedPhone = formatPhoneNumber(to);
+
+    console.debug(`yunpian-sms send to: ${formattedPhone}, messageContent: ${messageContent}`);
+
+    if (!enableInternational && formattedPhone.startsWith('+')) {
+      throw new ConnectorError(ConnectorErrorCodes.General, '暂不支持海外手机号');
```

Comment:
I think the error message is a little bit mis-leading, as `暂不支持海外手机号` means `International phone number is temporarily not supported`, but I think it would be more informative to ask the user to enable the international service in this case.
Also, we should use English as error message as well.

Reply:
I agree with you, but since this error will be exposed to the `end-user` frontend, it may need to be carefully considered.
Would it be fine to say: ‘The administrator has not enabled international SMS services’?